ui/imports/shared/views/AssetsView.qml

```diff
@@ -62,42 +69,78 @@ ColumnLayout {
             settingsKey: "WalletAssets"
         }
 
-        readonly property bool hasCommunityAssets: communityAssetsLV.count
-
         function hideAllCommunityTokens(communityId) {
-            const tokenSymbols = ModelUtils.getAll(communityAssetsLV.model, "symbol", "communityId", communityId)
+            const tokenSymbols = ModelUtils.getAll(regularAssetsLV.model, "symbol", "communityId", communityId)
             d.controller.settingsHideCommunityTokens(communityId, tokenSymbols)
         }
-    }
 
-    component CustomSFPM: SortFilterProxyModel {
-        id: customFilter
-        property bool isCommunity
+        readonly property SubmodelProxyModel assetsWithFilteredBalances: SubmodelProxyModel {
+            id: assetsWithFilteredBalances
+            sourceModel: root.assets
+            submodelRoleName: "balances"
+            delegateModel: SortFilterProxyModel {
+                sourceModel: submodel
+                filters: ExpressionFilter {
+                    expression: {
+                        root.networkFilters
+                        root.addressFilters
+                        return root.networkFilters.split(":").includes(chainId+"") &&
+                                (!!root.addressFilters ? root.addressFilters.toUpperCase() === account.toUpperCase() : true)
+                    }
+                }
+            }
+        }
 
-        sourceModel: root.assets
+        function getTotalBalance(balances, decimals) {
+            let totalBalance = 0
+            for(let i=0; i<balances.count; i++) {
+                totalBalance+=SQUtils.AmountsArithmetic.toNumber(ModelUtils.get(balances, i, "balance"), decimals)
+            }
+            return totalBalance
+        }
+    }
+
+    property SortFilterProxyModel customSFPM: SortFilterProxyModel {
+        sourceModel: d.assetsWithFilteredBalances
         proxyRoles: [
-            ExpressionRole {
-                name: "currentBalance"
-                expression: model.enabledNetworkBalance.amount
+                FastExpressionRole {
+                id: filter
+                 name: "currentBalance"
+                 expression: d.getTotalBalance(model.balances, model.decimals, root.addressFilters, root.networkFilters)
```

Comment:
What if the `balances` or the filters change? This being a JS function will not trigger reevaluation

Reply:
hey so, me adding the filters in this function call root.addressFilters and root.networkFilters re-triggers this expression role.